internal/pkg/config/config.go

```diff
@@ -228,22 +229,19 @@ var readConfigCached *Config
 var readConfigCachedErr error
 
 func ReadConfig() (*Config, error) {
-	loadedFromCache := true
 	readFileOnce.Do(func() {
-		loadedFromCache = false
 		c := new(Config)
 		bytes, err := ioutil.ReadFile(configFile)
 		if err != nil {
 			readConfigCachedErr = fmt.Errorf("error reading analysis config: %v", err)
 			return
 		}
 
-		if err := json.Unmarshal(bytes, c); err != nil {
+		if err := yaml.UnmarshalStrict(bytes, c); err != nil {
```

Comment:
I agree. One time I misspelled a field name in the JSON config and it failed silently, and I thought it was working because of the default "if field not present, match everything" behavior.